### ISSUE TITLE
TYPO3 9.5.1: Fix IsUniqueInSite arguments

### ISSUE THREAD

**Issue by nicozerr** [featdd/dpn_glossary#132]
Hi.

When I launch the wizard in order to update slug fields, this error occurs:
Uncaught TYPO3 Exception: Argument 2 passed to TYPO3\CMS\Core\DataHandling\SlugHelper::isUniqueInSite() must be an instance of TYPO3\CMS\Core\DataHandling\Model\RecordState, integer given, called in /var/www/html/public/typo3conf/ext/dpn_glossary/Classes/Updates/SlugUpdateWizard.php on line 79

Since [this commit](https://github.com/TYPO3-CMS/core/commit/d2950f72cec0d483b97a10bd45b8381475b560f8#diff-2d47aba761489522567e146d8ddc5379) on ext:core, please replace :

```
if (false === $slugHelper->isUniqueInSite($termSlug, $termUid, 0, $termLanguageUid)) {
```

by something like:

```
$state = RecordStateFactory::forName(Term::TABLE)->fromArray($term, $termPid, $termUid);
if (false === $slugHelper->isUniqueInSite($termSlug, $state)) {
```

In the file Classes/Updates/SlugUpdateWizard.php.
And this line too:

`use TYPO3\CMS\Core\DataHandling\Model\RecordStateFactory;`

**Comment by featdd** [comment]
Oh thought I could be a good guy giving a proper install wizard and then core just breaks the function 😞 

Will try to fix this as soon as I can, but If you already have a solution working and tested, please don't be shy and make a pull request.

Greetings
Daniel

**Comment by featdd** [comment]
merged #133 will make an update and push this to ter

**Comment by featdd** [comment]
Update [3.0.1](55c924bcd7069a12afd1a605af9b9a929607b8af) is up in TER

thanks for the pull request
